### PR TITLE
Fixes for NO_VS builds

### PR DESCRIPTION
**Pull request by psiborg112**
Fix build issues when NO_VS is set

Update CMakeLists.txt

Move all vs files into src/gui since they're not used by the command line version of the app

### CODE REVIEW

**Comment by ntonnaett** [comment]
We should have a GHA check for the NO_VS build. Doesn't need to be in this PR. Maybe @dyfer wants to help with that?

**Comment by psiborg112** [comment]
That would be useful to have down the line. Would like to clean up main a bit as well and encapsulate some of the VS specific stuff that seems to be happening there, but just wanted to get a working build to start with.

**Comment by mattahorton** [comment]
Good catch. This is not a build issue then? Because those tests pass.

**Comment by psiborg112** [comment]
We don't currently have any NO_VS builds in github actions that still include the rest of the GUI. So it wouldn't have been caught by the regular tests, just by me updating my build.

**Comment by mattahorton** [comment]
> We don't currently have any NO_VS builds in github actions that still include the rest of the GUI. So it wouldn't have been caught by the regular tests, just by me updating my build.

Gotcha. That makes total sense.

File: src/main.cpp [hunk 1]
```diff
@@ -314,9 +314,11 @@ int main(int argc, char* argv[])
         // Check if we need to show our first run window.
         QSettings settings;
         int uiMode = settings.value(QStringLiteral("UiMode"), QJackTrip::UNSET).toInt();
+#ifndef __unix__
```
Comment (mattahorton) [inline src/main.cpp:317]: is `__unix__` true for macOS?

File: src/gui/vsDevice.cpp [hunk 2]
```diff
@@ -474,13 +472,13 @@ QString VsDevice::randomString(int stringLength)
         "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789");
 
     if (!seeded) {
-        qsrand(QTime::currentTime().msec());
+        m_randomizer.seed((QTime::currentTime().msec()));
```
Comment (mattahorton) [inline src/gui/vsDevice.cpp:475]: Thanks for this. I wasn't sure how to update this one.

File: src/main.cpp (hunk 1 above)
Comment (psiborg112) [reply src/main.cpp:317]: No. But it is true on BSD which is why we're using it here instead of just using __linux__.

Comment (mattahorton) [reply src/main.cpp:317]: Thanks for the clarification!

**Review by mattahorton — APPROVED** [review]

**Comment by dyfer** [comment]
A follow up question: should we add no_vs build option to qmake and meson?

File: src/gui/vsDevice.cpp (hunk 2 above)
Comment (psiborg112) [reply src/gui/vsDevice.cpp:475]: No problem. There are more secure ways to do this without needing to seed things, but didn't know if you needed predictable results from the generator.

**Comment by psiborg112** [comment]
Yes, we would need a NO_VS build option on at least one of those. I'm 
happy to put one together for qmake when I get a chance.

**Comment by ntonnaett** [comment]
> Yes, we would need a NO_VS build option on at least one of those. I'm happy to put one together for qmake when I get a chance.

Meson and qmake already have a novs option.

**Comment by psiborg112** [comment]
CMake definitely would, because I did the initial VS development and that's what I use to build. I just forgot that it had been added to the other build systems. (Even though the commit history shows that I'm the one who added it to qmake.)

**Pull request merged**